io-engine/src/rebuild/rebuild_task.rs

```diff
@@ -65,11 +62,8 @@ impl RebuildTask {
         blk: u64,
         descriptor: &RebuildDescriptor,
     ) -> Result<bool, RebuildError> {
-        // Check if the block has to be copied.
-        if let Some(ref log) = self.rebuild_log {
-            if !log.is_modified(blk) {
-                return Ok(false);
-            }
+        if descriptor.is_blk_sync(blk) {
```

Comment:
did we ever test if there's any perf difference between walking the device range and skipping IO versus walking the log itself?

Reply:
No we didn't, as we don't have code walking the log in the first place :-)
I'd leave this optimization for future, as it would change the rebuild logic (tasks, jobs) considerably.